docs/chapters/traversal/index.md

```diff
@@ -0,0 +1,178 @@
+# Understand Jina Recursive Document Representation
+
+In Jina, each `Document` is represented as a recursive representation (tree).
+
+A rooted recursive representation has a root node and every node has X children.
+In Jina, the root node is the document itself, while the *left* & *right* children are referred to as *chunks* and *matches* respectively.
```

Comment:
Alex mentioned to capitalize Jina concepts, so I would use Document, Chunks, and Matches here.

Reply:
@yuanbit I capitalised `Document` and `Jina` since Document is a class, and made `matches` and `chunks` lower cases to make it consistency with our code example